controller.py

```diff
@@ -0,0 +1,182 @@
+                                8000, # HTTP access for LGTM Enterprise's startup log.
+                            ],
+                        },
+                    ],
+                },
+            },
+            # Firewall Rules for Workers
+            {
+                "name": deployment + "-workers",
+                "type": "compute.v1.firewall",
+                "properties": {
+                    "network": "$(ref." + network_name + ".selfLink)",
+                    "targetTags": [
+                        "lgtm-worker",
+                    ],
+                    "allowed": [
+                        {
+                            "IPProtocol": "tcp",
+                            "ports": [
+                                22, # Administrative SSH access to the virtual machine.
+                                8000, # HTTP access for LGTM Enterprise's startup log.
+                            ],
+                        },
+                    ],
+                },
+            },
+            # Firewall Rule for Internal Communication
+            {
+                "name": deployment + "-internal",
+                "type": "compute.v1.firewall",
+                "properties": {
+                    "network": "$(ref." + network_name + ".selfLink)",
+                    "allowed": [
+                        {
+                            "IPProtocol": "all",
+                        },
+                    ],
+                    "sourceRanges": [
+                        "10.128.0.0/9",
+                    ],
+                },
+            },
+            # Static IP for Controller
+            {
+                "name": ip_name,
+                "type": "compute.v1.address",
+                "properties": {
+                    "region": region,
+                },
+            },
+            # OS Disk for Controller
+            {
+                "name": os_disk_name,
+                "type": "compute.v1.disk",
+                "properties": {
+                    "zone": zone,
+                    "type": "zones/" + zone + "/diskTypes/pd-ssd",
+                    "sourceImage": _IMAGE,
+                },
+            },
+            # Data Disk for Controller
+            {
+                "name": deployment + "-data",
+                "type": "compute.v1.disk",
+                "properties": {
+                    "zone": zone,
+                    "type": "zones/" + zone + "/diskTypes/pd-ssd",
+                    "sizeGb": context.properties["data-disk-size-gb"],
+                },
+            },
+            # Controller Instance
+            {
+                "name": deployment,
+                "type": "compute.v1.instance",
+                "properties": {
+                    "zone": zone,
+                    "machineType": "zones/" + zone + "/machineTypes/" + context.properties["virtual-machine-size"],
+                    "hostname": deployment + "." + zone + ".c.semmle-oss-testing.internal",
```

Comment:
Should we be referencing `semmle-oss-testing` here?

Reply:
Ah, no. That should be the current project.